kikuchipy/indexing/merge_crystalmaps.py

```diff
@@ -0,0 +1,140 @@
+    Both props are merge sorted into the returned map.
+
+    Parameters
+    ----------
+    xmaps : List[CrystalMap]
+        List of `CrystalMap`s with "template_indices" and "metric_results" in prop
+    mean_n_largest : int, optional
+        Number of top metric results to take the mean of,
+        before comparing and put in the property `merged_top_metric_result`, by default 1
+    metric : Union[str, SimilarityMetric], optional
+        Similarity metric, by default None
+
+    Returns
+    -------
+    merged_xmap : CrystalMap
+        A CrystalMap with added prop "merged_top_metric_result"
+
+    Notes
+    -----
+    `mean_n_largest` can be given with a negative sign if metric is not given
+    in order to choose the lowest valued metric results.
+
+    Raises
+    ------
+    ValueError
+        If `xmaps` contains CrystalMaps with equal phases.
+    """
+    if metric is None:
+        sign = copysign(1, mean_n_largest)
+        mean_n_largest = abs(mean_n_largest)
+    else:
+        sign = SIMILARITY_METRICS.get(metric, metric).sign
+
+    top_results_across_xmaps = np.array(
+        [
+            np.mean(xmap.metric_results[:, :mean_n_largest], axis=1)
+            for xmap in xmaps
+        ]
+    )
+
+    phase_id = np.argmax(sign * top_results_across_xmaps, axis=0)
+    merged_top_metric_result = np.choose(phase_id, top_results_across_xmaps)
+
+    metric_results = np.concatenate(
+        [xmap.metric_results for xmap in xmaps], axis=1
+    )
+
+    metric_result_sorted_indicies = np.argsort(
+        sign * -metric_results, kind="mergesort", axis=1
+    )
+
+    template_indices = np.concatenate(
+        [xmap.template_indices for xmap in xmaps],
+        axis=1,
+    )
+    template_indices = np.take_along_axis(
+        template_indices, metric_result_sorted_indicies, axis=1
+    )
+    metric_results = np.take_along_axis(
+        metric_results, metric_result_sorted_indicies, axis=1
+    )
+
+    prop = {
+        "merged_top_metric_result": merged_top_metric_result,
+        "template_indices": template_indices,
+        "metric_results": metric_results,
+    }
+
+    # The rotations can be examined more carefully, takes now all keep_n only for the top 1 phase
+    # Maybe not ideal
+    rotations_across_xmaps = np.array([xmap.rotations.data for xmap in xmaps])
+    rotations = np.array(
+        [rotations_across_xmaps[id][i] for i, id in enumerate(phase_id)]
+    )
+    rotations = Rotation(rotations)
+
+    # OBS! Can raise ValueError if phase already in phase_lists
+    phase_list = PhaseList()
+    for xmap in xmaps:
+        try:
+            phase_list.add(xmap.phases_in_data)
+        except ValueError:
+            warnings.warn(
+                f"One or more of the phases in {xmap.phases_in_data} are already"
+                "in the PhaseList.",
+                UserWarning,
+            )
+
+    merged_xmap = CrystalMap(
+        rotations,
+        phase_id=phase_id,
```

Comment:
By the way, original phase IDs are lost when adding phases to a new PhaseList (https://github.com/pyxem/orix/blob/v0.5.0/orix/crystal_map/phase_list.py#L642-L693). Pretty sure we must handle this correctly. If it's okay with you @onatlandsmyr, I can do this.

Reply:
Currently if two maps with the same phase names are being merged phase_list will have length 1 while phase_id have length 2, might that be problematic? If so, maybe a solution can be to set phase_list=None on conflicts like this. 

I don't see why phase_id must be preserved or how it even can be; usually it's of length 1 with value 0 for both/all maps to be merged.

Reply:
> Currently if two maps with the same phase names are being merged phase_list will have length 1 while phase_id have length 2, might that be problematic? If so, maybe a solution can be to set phase_list=None on conflicts like this.

`CrystalMap` requires as many phases in the `PhaseList` as there are unique integers in `phase_id`:
* https://orix.readthedocs.io/en/stable/reference.html#orix.crystal_map.crystal_map.CrystalMap.__init__
* https://github.com/pyxem/orix/blob/v0.5.0/orix/crystal_map/crystal_map.py#L148

While `CrystalMap.rotations` can have `n` rotations per point (best, 2nd best, etc.), only one `phase_id` per point is supported, so the list of best rotations must belong to the same phase.

> I don't see why phase_id must be preserved or how it even can be; usually it's of length 1 with value 0 for both/all maps to be merged.

I agree that the phase ID in general cannot be preserved, so:
* new unique IDs must be assigned to each unique phase
* the `phase_id` of points in the map where this phase's solution is the best one must be set to this new ID